specs/protocol/jovian/exec-engine.md

```diff
@@ -0,0 +1,70 @@
+# Jovian: Execution Engine
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Minimum Base Fee](#minimum-base-fee)
+  - [Minimum Base Fee in Block Header](#minimum-base-fee-in-block-header)
+  - [Minimum Base Fee in `PayloadAttributesV3`](#minimum-base-fee-in-payloadattributesv3)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Minimum Base Fee
+
+Jovian introduces a configurable minimum base fee to reduce the duration of priority-fee auctions on OP Stack chains.
+
+The minimum base fee is configured via `SystemConfig` (see `./system-config.md`) and enforced by the execution engine
+via the block header `extraData` encoding and the Engine API `PayloadAttributesV3` parameters.
+
+### Minimum Base Fee in Block Header
+
+Like [Holocene's dynamic EIP-1559 parameters](../holocene/exec-engine.md#dynamic-eip-1559-parameters), Jovian encodes
+fee parameters in the `extraData` field of each L2 block header. The format is extended to include an additional byte
+for the minimum base fee exponent.
+
+`extraData` layout (10 bytes total):
+
+| Name             | Type               | Byte Offset |
+| ---------------- | ------------------ | ----------- |
+| `version`        | `u8`               | `[0, 1)`    |
+| `denominator`    | `u32 (big-endian)` | `[1, 5)`    |
+| `elasticity`     | `u32 (big-endian)` | `[5, 9)`    |
+| `minBaseFeeLog2` | `u8`               | `[9, 10)`   |
+
+Constraints:
+
+- `version` MUST be `1` (incremented from Holocene's `0`).
+- `denominator` MUST be non-zero.
+- There MUST NOT be any data beyond these 10 bytes.
+
+The `minBaseFeeLog2` field encodes the base-2 logarithm of the minimum base fee in wei. A value of `0` disables the
+minimum base fee entirely.
```

Comment:
I think some pseudo-code here would be good:
```
	if minBaseFeeLog2 > 0 {
		minBaseFee = 2**minBaseFeeLog2
		if baseFee < minBaseFee {
			baseFee = minBaseFee
		}
	}
```

Reply:
08124edff723c0b1eb87ef42491e835bff1a947e